src/lib/mdns/ServiceNaming.cpp

```diff
@@ -67,43 +48,34 @@ CHIP_ERROR ExtractIdFromInstanceName(const char * name, PeerId * peerId)
     ReturnErrorCodeIf(name == nullptr, CHIP_ERROR_INVALID_ARGUMENT);
     ReturnErrorCodeIf(peerId == nullptr, CHIP_ERROR_INVALID_ARGUMENT);
 
-    peerId->SetNodeId(0);
-    peerId->SetFabricId(0);
-
-    bool deliminatorFound = false;
-    bool hasFabricPart    = false;
-    bool hasNodePart      = false;
-
-    for (; *name != '\0'; name++)
-    {
-        if (*name == '.')
-        {
-            break;
-        }
-        else if (*name == '-')
-        {
-            deliminatorFound = true;
-            continue;
-        }
-
-        uint8_t val = HexToInt(*name);
-        ReturnErrorCodeIf(val == UINT8_MAX, CHIP_ERROR_WRONG_NODE_ID);
-
-        if (deliminatorFound)
-        {
-            hasNodePart = true;
-            peerId->SetNodeId(peerId->GetNodeId() * 16 + val);
-        }
-        else
-        {
-            hasFabricPart = true;
-            peerId->SetFabricId(peerId->GetFabricId() * 16 + val);
-        }
-    }
-
-    ReturnErrorCodeIf(!deliminatorFound, CHIP_ERROR_WRONG_NODE_ID);
-    ReturnErrorCodeIf(!hasNodePart, CHIP_ERROR_WRONG_NODE_ID);
-    ReturnErrorCodeIf(!hasFabricPart, CHIP_ERROR_WRONG_NODE_ID);
+    // Make sure the string is long enough.
+    static constexpr size_t fabricIdByteLength   = 8;
+    static constexpr size_t fabricIdStringLength = fabricIdByteLength * 2;
+    static constexpr size_t nodeIdByteLength     = 8;
+    static constexpr size_t nodeIdStringLength   = nodeIdByteLength * 2;
+    static constexpr size_t totalLength          = fabricIdStringLength + nodeIdStringLength + 1; // +1 for '-'
+    size_t len                                   = strlen(name);
```

Comment:
Suggest `strnlen(name, totalLength + 1)` and check that `len != totalLength` in the next line

Reply:
Good call on the `strnlen`; I had idly wondered how best to avoid walking the whole string if it happens to be long...

But `len != totalLength` does not work because we _could_ have `len > totalLength` if our instance name is followed by `'.'`.  So going to do this:
```
    // Ensure we have at least totalLength chars.
    size_t len = strnlen(name, totalLength);
    ReturnErrorCodeIf(len < totalLength, CHIP_ERROR_INVALID_ARGUMENT);
```